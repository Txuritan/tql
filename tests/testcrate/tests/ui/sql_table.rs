/*
 * Copyright (c) 2017 Boucher, Antoni <bouanto@zoho.com>
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy of
 * this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to
 * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
 * the Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
 * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
 * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
 * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
 * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

//! Tests of the `#[SqlTable]` attribute.

#![feature(proc_macro)] // FIXME: bad span for field nested_options in stderr (should be on Option<String>, not just Option).

#[macro_use]
extern crate tql_macros;

struct Connection {
    value: String,
}

#[derive(SqlTable)]
struct Table<'a> {
    //~^ WARNING No primary key found
    string: &'a str,
    //~^ ERROR use of unsupported type name `& 'a str`
    connection: Connection,
    //~^ ERROR use of unsupported type name `Connection`
    connection2: Option<Connection>,
    //~^ ERROR use of unsupported type name `Connection`
    nested_options: Option<Option<String>>,
    //~^ ERROR use of unsupported type name `Option<String>`
    datetime: DateTime,
    //~^ ERROR use of unsupported type name `DateTime`
    datetime_i32: DateTime<i32>,
    //~^ ERROR use of unsupported type name `DateTime<i32>`
    foreign_value: ForeignKey,
    //~^ ERROR use of unsupported type name `ForeignKey`
    optional_value: Option,
    //~^ ERROR use of unsupported type name `Option`
    vector: Vec,
    //~^ ERROR use of unsupported type name `Vec`
    vector_i32: Vec<i32>,
    //~^ ERROR use of unsupported type name `Vec<i32>`
}
