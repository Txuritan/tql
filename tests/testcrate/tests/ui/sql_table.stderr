warning: No primary key found
  --> $DIR/sql_table.rs:34:8
   |
34 | struct Table<'a> {
   |        ^^^^^

error: use of unsupported type name `& 'a str`
  --> $DIR/sql_table.rs:36:13
   |
36 |     string: &'a str,
   |             ^^^^^^^

error: use of unsupported type name `Connection`
  --> $DIR/sql_table.rs:38:17
   |
38 |     connection: Connection,
   |                 ^^^^^^^^^^

error: use of unsupported type name `Connection`
  --> $DIR/sql_table.rs:40:25
   |
40 |     connection2: Option<Connection>,
   |                         ^^^^^^^^^^

error: use of unsupported type name `Option<String>`
  --> $DIR/sql_table.rs:42:28
   |
42 |     nested_options: Option<Option<String>>,
   |                            ^^^^^^^^^^^^^^^

error: use of unsupported type name `DateTime`
  --> $DIR/sql_table.rs:44:15
   |
44 |     datetime: DateTime,
   |               ^^^^^^^^

error: use of unsupported type name `DateTime<i32>`
  --> $DIR/sql_table.rs:46:19
   |
46 |     datetime_i32: DateTime<i32>,
   |                   ^^^^^^^^^^^^^

error: use of unsupported type name `ForeignKey`
  --> $DIR/sql_table.rs:48:20
   |
48 |     foreign_value: ForeignKey,
   |                    ^^^^^^^^^^

error: use of unsupported type name `Option`
  --> $DIR/sql_table.rs:50:21
   |
50 |     optional_value: Option,
   |                     ^^^^^^

error: use of unsupported type name `Vec`
  --> $DIR/sql_table.rs:52:13
   |
52 |     vector: Vec,
   |             ^^^

error: use of unsupported type name `Vec<i32>`
  --> $DIR/sql_table.rs:54:17
   |
54 |     vector_i32: Vec<i32>,
   |                 ^^^^^^^^

error: aborting due to 10 previous errors

